File: sentimental/readability/readability.py
from cs50 import get_string

#getting promp from user


text = get_string("Text: ")

# counting the letters in text


def count_letters(letters):
    no_letters = 0
    for character in letters:
        if character.isalpha():
            no_letters += 1
    return no_letters


letters = count_letters(text)

# counting the words in text


def count_words(words):
    words = words.split()
    no_words = len(words)
    return no_words
 
 
words = count_words(text)

# counting the number of snetences in text


def count_sentences(sentences):
    no_sentences = sentences.count(".") + sentences.count("!") + sentences.count ("?")
    return no_sentences


sentences = count_sentences(text)
# checking Coleman-Liau index


l = letters / words * 100
s = sentences / words * 100
index = 0.0588 * l - 0.296 * s - 15.8
#rounding the value 

grade = round(index)
#printing the grade according to index

if grade >= 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print(f"Grade {grade}")

File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int main(void)
{
    // prompt the user for some text
    string text = get_string("Text: ");
    // count the numbers of letters, words and sentences
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    // compute the coleman-liau index
    float L = (float) letters / words * 100;
    float S = (float) sentences / words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    // round the index to the nearest whole number
    int grade = round(index);
    // print the grade level
    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}
int count_letters(string text)
{
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}
int count_words(string text)
{
    int count = 1;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isspace(text[i]))
        {
            count++;
        }
    }
    return count;
}
int count_sentences(string text)
{
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count++;
        }
    }
    return count;
}